coverage.py

```diff
@@ -43,50 +45,50 @@
 
 from mirror import ImageScaler
 
-from oclc import LinkedDataCoverageProvider
+from oclc import (
+    LinkedDataCoverageProvider,
+)
 
 from viaf import (
     VIAFClient, 
 )
 
 
-class IdentifierResolutionCoverageProvider(CoverageProvider):
-    """ Resolve all of the Identifiers with CoverageProviders in transient 
-    failure states, turning them into Editions with LicensePools.
-    Create CoverageProviders to contact 3rd party entities for information on 
-    Identifier-represented library item (book).
+class IdentifierResolutionCoverageProvider(CatalogCoverageProvider):
+    """Make sure all Identifiers registered as needing coverage by this
+    CoverageProvider become Works with Editions and (probably dummy)
+    LicensePools.
+
+    Coverage happens by running the Identifier through _other_
+    CoverageProviders, filling in the blanks with additional data from
+    third-party entities.
 
-    For ISBNs, make a bunch of Resources, rather than LicensePooled Editions.
+    For ISBNs, we end up with a bunch of Resources, rather than
+    Works. TODO: This needs to change.
     """
 
-    CAN_CREATE_LICENSE_POOLS = True
+    SERVICE_NAME = "Identifier Resolution Coverage Provider"
+    DATA_SOURCE_NAME = DataSource.INTERNAL_PROCESSING
+    INPUT_IDENTIFIER_TYPES = [Identifier.OVERDRIVE_ID, Identifier.ISBN]
+    OPERATION = CoverageRecord.RESOLVE_IDENTIFIER_OPERATION
+    
     LICENSE_SOURCE_NOT_ACCESSIBLE = (
         "Could not access underlying license source over the network.")
     UNKNOWN_FAILURE = "Unknown failure."
 
-    def __init__(self, _db, batch_size=10, cutoff_time=None,
-                 uploader=None, providers=None, **kwargs):
-        output_source, made_new = get_one_or_create(
-            _db, DataSource,
-            name=DataSource.INTERNAL_PROCESSING
-        )
-        # Other components don't have INTERNAL_PROCESSING as offering
-        # licenses, but we do, because we're responsible for managing
-        # LicensePools.
-        output_source.offers_licenses=True
-        input_identifier_types = [Identifier.OVERDRIVE_ID, Identifier.ISBN]
+    def __init__(self, collection, uploader=None,
+                 viaf_client=None, linked_data_coverage_provider=None,
+                 overdrive_api_class=OverdriveAPI,
+                 **kwargs):
```

Comment:
We're now taking a lot of arguments that let us plug in mock APIs, which wasn't possible before.